ush/extractvars_tools.sh

```diff
@@ -0,0 +1,43 @@
+#! /usr/bin/env bash
+
+gen_parmlist() {
+  infileg=$1
+  requestedvar_in_file1=$2
+  varlist=$3
+  rm -vf "${requestedvar_in_file1}"
+  while read -r vari; do
+    varinfile=$(${WGRIB2} "${infileg}" | grep "${vari}") || true
+    if [[ -n "${varinfile}" ]];then # if varinfile is not empty
+      echo "${vari}" >> "${requestedvar_in_file1}"
+    fi
+  done <"${varlist}"
+}
+
+check_atmos() {
+  requestedvar_in_file1=$1
+  requestedvar_in_file2=$2
+  requestedvar_in_allgrb2file=requestedvar_in_allgrb2file.txt
+  rm -rvf "${requestedvar_in_allgrb2file}"
+  cat "${requestedvar_in_file1}" "${requestedvar_in_file2}" >> "${requestedvar_in_allgrb2file}"
+  mapfile -t requestedvar_in_allgrb2file_arr < "${requestedvar_in_allgrb2file}"
+  while read -r vari; do
+    if [[ ! ${requestedvar_in_allgrb2file_arr[*]} =~ ${vari} ]] ;then
+      echo "WARNING: PARM VARIABLE (${vari}) is not available in pgrb and pgrb2b."
+    fi
+  done <"${varlist}"
```

Comment:
Where is `varlist` defined?  Is this supposed to be the `varlist` variable sent to `gen_parmlist` or the `varlist` in `ush/gefs_<component>_extractvars.sh`?  Suggest taking `varlist` as an argument to the `check_atmos` function to avoid ambiguity.

Reply:
`varlist` is defined in `ush/gefs_atmos_extractvars.sh`. It is a parm file that includes a list of variables. I will take `varlist` in as an argument to the `check_atmos` function, just like I am with the `gen_parmlist` function. `varlist` in the `check_atmos` function is the same `varlist` in the `gen_parmlist` function.

Reply:
Done.

Reply:
I would suggest changing the name of `varlist` to something different in this function to prevent redefining `varlist` in `gefs_atmos_extractvars.sh`.  It doesn't matter too much in this case since `varlist` is being set to itself, but something to be mindful of.  Here is an example of a potentially problematic situation:

```bash
alpha() {
  a=$1
}
export a=3
alpha "not 3"
echo $a
# Echoes "not 3"
```